packages/mui-joy/src/CssBaseline/CssBaseline.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { GlobalStyles } from '@mui/system';
+import { Theme, DefaultColorScheme, ColorSystem } from '../styles/types';
+import { CssBaselineProps } from './CssBaselineProps';
+
+/**
+ * Kickstart an elegant, consistent, and simple baseline to build upon.
+ */
+function CssBaseline({ children, disableColorScheme = false }: CssBaselineProps) {
+  return (
+    <React.Fragment>
+      <GlobalStyles
+        styles={(theme: Theme) => {
+          const colorSchemeStyles: Record<string, any> = {};
+          if (!disableColorScheme) {
+            // The CssBaseline is wrapped inside a CssVarsProvider
+            (
+              Object.entries(theme.colorSchemes) as Array<[DefaultColorScheme, ColorSystem]>
+            ).forEach(([key, scheme]) => {
+              colorSchemeStyles[theme.getColorSchemeSelector(key).replace(/\s*&/, '')] = {
```

Comment:
Can you clarify what is this doing?
```
theme.getColorSchemeSelector(key).replace(/\s*&/, '')
```

We have it in many places, maybe we should extract the logic?

Reply:
`theme.getColorSchemeSelector(key)` returns a string, `[data-joy-color-scheme="$key"] &`, with an ` &` at the end so that developers can use it to write custom styles in `styled` or `sx` like this:

```js
styled('div')(({ theme }) => ({
  [theme.getColorSchemeSelector('dark')]: { ... }
  // easier than writing
  // [`${theme.getColorSchemeSelector('dark')} &`]: { ... }
}))
```

---

```
theme.getColorSchemeSelector(key).replace(/\s*&/, '') // '[data-joy-color-scheme="dark"]'
```

I think this case would be rare but I could provide the `colorSchemeSelector` from the theme to be used like:

```js
[`[${theme.colorSchemeSelector}="dark"]`]: { ... }
```

Reply:
Thinking again, I rather go with `theme.getColorSchemeSelector(key).replace(/\s*&/, '')` for now because I don't want to add a new property to the theme. Let's wait to see how the community would like to use the API.